STDEV-3264

apliter opened the pull request:
Added ACTION_DISCOUNT_SCREEN_ADDITIONAL_ITEMS for Buy\Buyback receipt
Added receiptDiscountEvent for Buy\Buyback receipt

Commit:
STDEV-3264
Added ACTION_DISCOUNT_SCREEN_ADDITIONAL_ITEMS for Buy\Buyback receipt
Added receiptDiscountEvent for Buy\Buyback receipt

Commit:
STDEV-3264
javadoc correction

flooduk commented on build.gradle:
```diff
@@ -6,15 +6,15 @@ apply plugin: 'maven-publish'
         targetSdkVersion 25
         versionCode version
-        versionName "0.4.5"
+        versionName "STDEV-1901"
```
ой

Commit:
STDEV-3264
build.gradle correction

Commit:
STDEV-3264
build.gradle correction

Commit:
STDEV-3264
Added paybackBySum\buybackBySum events and integration services.

xcam commented on src/main/java/ru/evotor/framework/receipt/formation/event/handler/service/BuyIntegrationService.kt:
```diff
@@ -1,23 +1,29 @@
     }
 
+    @RequiresIntentAction(ACTION_DISCOUNT_SCREEN_ADDITIONAL_ITEMS)
+    open fun handleEvent(event: DiscountScreenAdditionalItemsEvent): IBundlable? = null
```
что я должен вернуть из этого метода? Какие данные?

apliter replied:
+

xcam commented on src/main/java/ru/evotor/framework/core/action/event/receipt/discount/ReceiptDiscountEvent.java:
```diff
@@ -32,6 +32,34 @@
+     * <p>
+     * Значение константы: <code>evo.v2.receipt.paybackBySum.receiptDiscount</code>.
+     */
+    public static final String NAME_PAYBACK_BY_SUM_RECEIPT = "evo.v2.receipt.paybackBySum.receiptDiscount";
```
понятие BySum думаю не стоит вводить. Скорее стоит обособить возврат на основании

apliter replied:
Нельзя так. 
Причина: В возврате на основании чека не должны быть доступны интеграции ReceiptDiscountEvent, а в возврате без чека (BY_SUM) интеграции должны быть видны. 
Чем плохо разделение возвратов?

apliter replied:
+

xcam reviewed (changes requested):
см комментарии

Commit:
STDEV-3264
Second iteration.
Added ReceiptDiscountEvents:
NAME_PAYBACK_WITH_SELL_RECEIPT
NAME_BUYBACK_WITH_BUY_RECEIPT

Commit:
STDEV-3264
Added return result in ACTION_DISCOUNT_SCREEN_ADDITIONAL_ITEMS handle.

Commit:
STDEV-3264
codestyle

Commit:
STDEV-3369
Added medicine product type.

flooduk commented on src/main/java/ru/evotor/framework/core/action/event/receipt/discount/ReceiptDiscountEvent.java:
```diff
@@ -26,12 +26,40 @@
+    public static final String NAME_PAYBACK_WITH_SELL_RECEIPT = "evo.v2.receipt.paybackWithSell.receiptDiscount";
+
+    /**
+     * Скидка начислена на чек покупки без основания.
```
на чек покупки. "без основания" - лишнее

apliter replied:
+

apliter replied:
исправил

xcam commented on src/main/java/ru/evotor/framework/inventory/ProductType.java:
```diff
@@ -33,5 +33,10 @@
+    /**
+     * Маркированные лекарства
+     */
+    MEDICINE_MARKED
```
Это как сюда попало?

apliter replied:
откатил

xcam commented on src/main/java/ru/evotor/framework/receipt/Position.java:
```diff
@@ -869,7 +869,21 @@ public Builder toShoesMarked(
+            return this;
+        }
+
+        public Builder toMedicineMarked(
```
это как сюда попало?

xcam commented on src/main/java/ru/evotor/framework/receipt/Position.java:
```diff
@@ -907,11 +921,7 @@ private void setAlcoParams(
-        }
-
-        private void setShoesParams(String mark) {
+        private void setMarkedParams(String mark) {
```
а когда добавятся параметры рецепта для лекарств не придется обратно все откатывать?

apliter replied:
вернул

xcam commented on src/main/java/ru/evotor/framework/receipt/formation/event/handler/service/PaybackIntegrationService.kt:
```diff
@@ -19,12 +18,11 @@ abstract class PaybackIntegrationService : IntegrationServiceV2() {
 
     @RequiresIntentAction(ACTION_DISCOUNT_SCREEN_ADDITIONAL_ITEMS)
-    open fun handleEvent(event: DiscountScreenAdditionalItemsEvent): IBundlable? = null
+    open fun handleEvent(event: DiscountScreenAdditionalItemsEvent): DiscountScreenAdditionalItemsEvent? = null
```
т.е. я на входе получаю event и вернуть должен event? Кажется мне, что все таки ничего не надо возвращать

apliter replied:
так как onEvent, в котором дергается две реализации handleEvent имеет тип возвращаемого результата
ReturnPurchaserRequisitesForPrintGroupRequestedEvent.Result? то во избежании проблем 
handleEvent для ACTION_DISCOUNT_SCREEN_ADDITIONAL_ITEMS, который не имеет результата, будет возвращать Nothing? = null

Commit:
STDEV-3369
revert added medicine

Commit:
STDEV-3264
revert added medicine

Commit:
STDEV-3264
edited javadoc

Commit:
STDEV-3264
change handleEvent return type for Nothing?
